edx_django_utils/cache/__init__.py

```diff
@@ -3,9 +3,6 @@
 
 See README.rst for details.
 """
+from __future__ import absolute_import, unicode_literals
 
-from .utils import (
-    DEFAULT_REQUEST_CACHE,
-    RequestCache,
-    TieredCache
-)
+from .utils import DEFAULT_REQUEST_CACHE, RequestCache, TieredCache, get_user
```

Comment:
I relate to @robrap's feeling of wrong-ness.

We need to clarify the following in this repo's `README` file:
The purpose of `edx-django-utils` is to provide a common library across our Django-based microservices with architecturally significant utilities that we believe are necessary to be consistent across all IDAs.  We prefer not to include utilities that will unnecessarily tightly couple IDAs unless there's a strong reason for consistency or DRY-ness.

Regarding the `get_user` functionality, in particular, to me, it seems like a specific performance optimization that may not generalize to most needs.  When optimizing SQL queries for the `User` object, many callers will want to prefetch other user-related tables along with the user data.  Also, caching comes with the additional complexity of cache-invalidation, which callers of `get_user` will need to ensure happens appropriately by all of its own callers - especially in tests and in bulk-user operations.

Reply:
The impulse for this PR came from noticing a few spots where the code in one module passes the user_id to another module, and that one queries for the User (without doing any select_related stuff, either). So, it seems obvious to me to either change the callers to send user objects instead of user ids, or else cache these lookups. In fact, most of the time, they're probably querying for the currently logged in user, which is already available on the request. It would be good to clean up this pattern. Invalidation isn't a concern, either, in the instances I've seen, and this cache is scoped on the request, anyway.

But if there's resistance to adding a function here, I'll just close this PR. It's not worth the effort.

Reply:
Ah.  Thanks for the clarification.  In that case, there's already a utility method for the code to use: [crum.get_current_user](https://github.com/ninemoreminutes/django-crum/blob/87f482f01a4524dd480d38e18d163eb1a6aa5ba3/crum/__init__.py#L45).  Some of our IDAs, including edx-platform, have installed the [Current Request User Middleware](https://github.com/ninemoreminutes/django-crum).  So you can use it to get the currently logged in user.

Reply:
@nasthagiri The users are not necessarily the current user -- for instance, there can be a masquerading user.

Reply:
Then you can use or create helpers in masquerade.py for getting the correct user for the request, right?  This allows you to decide whether to use the requesting-user or the masquerading-user.

Reply:
I think you're missing the point of this function. There's code that passes around user_ids for whatever reason, but at some point a function further down the line needs access to the user object, so we see `User.objects.get(pk=user_id)`. The original calling function may have already had the user object in scope, and it may even be the currently logged in user. 

Now, maybe it makes sense to change all of these functions to pass around user objects instead of user ids, but I suspect there would be objection to that approach, too.

Reply:
Yup, I understand.  I was proposing a solution that you can implement and support within edx-platform itself.  Getting the current user is a common need - and if edx-platform has a concept of a masquerading user, then having a helper method to get the current user off of that module makes sense.

Beyond the need of getting the current user, you have to be extra careful of low-level functionality calling such a `get_user` method, if the low-level function is called in a bulk operation that retrieves many users within the request.

Regardless, I still question the value of having a common caching utility like this that we need to support and maintain across our IDAs.  Getting a `user` object is an IDA-specific activity, with each IDA having its own requirements and evolving to its own needs.

Yes, I prefer passing the `User` object around rather than passing the `user_id`.  The only exception to this is when crossing process boundaries - for example when calling a celery method or an API.